### DISCUSSION TITLE
I can't log in to Frepple from the odoo frame

### DISCUSSION THREAD

**Discussion by Icon003** [frePPLe/frepple#392] [Q&A]
I will deploy Odoo and Frepple on different machines, set up interaction with them according to the documentation: https://frepple.org/docs/current/integration-guide/odoo-connector.php#installation-and-configuration . Frepple appeared in odoo, but I had to add in djangosetting.py X_FRAME_OPTIONS = "ALLOWALL" because I caught an error related to the prohibition of displaying Frepple in the frame. Frepple displayed the login page in the Odoo frame, but I can't log in, I enter the username and password, and the page is updated and again offers to log in, and so on endlessly.

Please tell me what the problem is, I am grateful in advance.

**Comment by Icon003** [comment]
I realized from the logs that I was getting CSRF token errors

[Tue Aug 17 22:40:49.907843 2021] [wsgi:error] [pid 30:tid 140580563961600] [remote 37.214.48.243:54813] ERROR Invalid user in webtoken
[Tue Aug 17 22:40:55.537201 2021] [wsgi:error] [pid 31:tid 140580563961600] [remote 37.214.48.243:30755] ERROR CSRF failure detected
[Tue Aug 17 22:40:55.537980 2021] [wsgi:error] [pid 31:tid 140580563961600] [remote 37.214.48.243:30755] INFO Forbidden (CSRF cookie not set.): /data/login/

How can I avoid this?

**Comment by jdetaeye** [comment]
Does your odoo user match the user name in frepple? The first line in the error suggests that not the case.

**Reply by Icon003** [reply]
I don't understand what you are talking about, but why would a user match in two different systems? I'm trying to log in to frepple with the standard admin admin under which it goes if I don't use frame

**Reply by jdetaeye** [reply]
The odoo integration only deals the "authentication" in frepple (ie "who is this user?"), but not the "authorization" in frepple (ie "what is this user allowed to do in frepple?").    See https://www.google.com/search?q=authentication+vs+authorization for more info

The user needs to be defined in both odoo and frepple with the same name.   In this way you can configure the authorization of the user completely on frepple side.

**Reply by Icon003** [reply]
I have configured authorization, frepple does not want to authorize due to a csfr token error. I have configured authorization, frepple does not want to authorize due to a csfr token error. I wrote this in the second issues.

[Wed Aug 18 06: 34: 25.041670 2021] [wsgi: error] [pid 24:tid 140236854331136] [deleted 37.214.48.243:20758] ERROR CSRF error detected
[Wed Aug 18 06: 34: 25.042667 2021] [wsgi: error] [pid 24:tid 140236854331136] [deleted 37.214.48.243:20758] INFORMATION is prohibited (CSRF cookie is not installed.): / data/login/
